### PR TITLE
src/components/form: add component FormCoordinatorApplication to display on company coordinator page

### PR DESCRIPTION
**Pull request by maceksimon**
Add `FormCoordinatorApplication` to be displayed on the company coordinator page.

* Use required text field and phone field
* Enable custom labels for the phone field to fit this use case
* Add checkboxes for privacy policy and challenge terms
* Update translation keys to better reflect content in case of `labelPrivacyConsent`
* Add component tests

TODO (in separate PR): create a route for data privacy protection page and add it to checkbox field.

Screenshot 2024-06-10 at 13 33 28

### CODE REVIEW

**Comment by maceksimon** [comment]
Undefined error in test.

Screenshot 2024-06-12 at 16 08 53

**Review by tmszi — COMMENTED** [review]
Is your phone form field required field or not?

File: src/components/form/FormCoordinatorApplication.vue [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+</script>
+
+<template>
+  <q-form
+    ref="formCoordinatorApplicationRef"
+    data-cy="form-coordinator-application"
+  >
+    <div class="row q-col-gutter-md">
+      <div class="col-12 col-sm-6">
+        <!-- Input: Company name -->
```
Comment (tmszi) [inline src/components/form/FormCoordinatorApplication.vue:79]: Company job position

File: src/components/form/FormCoordinatorApplication.vue [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        <form-field-text-required
+          v-model="position"
+          name="name"
+          label="form.labelYourPosition"
+          data-cy="form-coordinator-position"
+        />
+      </div>
+      <div class="col-12 col-sm-6">
+        <form-field-phone
```
Comment (tmszi) [inline src/components/form/FormCoordinatorApplication.vue:88]: Missing comment `<!-- Input: Your telephone number -->`

File: src/components/form/FormCoordinatorApplication.vue [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+          v-model="phone"
+          name="phone"
+          label="form.labelYourPhone"
+          data-cy="form-coordinator-phone"
+        />
+      </div>
+      <div class="col-12">
```
Comment (tmszi) [inline src/components/form/FormCoordinatorApplication.vue:95]: Missing comment

**Comment by tmszi** [comment]
> Undefined error in test.
> Screenshot 2024-06-12 at 16 08 53

I have increased the time limit of the GitHub actions to 2 hours https://github.com/auto-mat/ride-to-work-by-bike-frontend/commit/be5f4159cced9c17bf6d6d1b520fbf5d92c04297. Could you rebase this PR to include this fix, please? If this repair will work you could rebase remaining PRs.

File: src/components/form/FormCoordinatorApplication.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/form/FormCoordinatorApplication.vue:89]: Added.

File: src/components/form/FormCoordinatorApplication.vue [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            data-cy="form-responsibility-input"
+          >
+            <!-- Default slot: label -->
+            <span>
+              {{ $t('form.labelCoordinatorResponsibility') }}
+            </span>
+          </q-checkbox>
+        </q-field>
+      </div>
+      <div class="col-12">
```
[the remark this answers is not in the document]
Comment (maceksimon) [reply src/components/form/FormCoordinatorApplication.vue:124]: Added.

File: src/components/form/FormCoordinatorApplication.vue (hunk 3 above)
Comment (maceksimon) [reply src/components/form/FormCoordinatorApplication.vue:96]: Added.

File: src/components/form/FormCoordinatorApplication.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/form/FormCoordinatorApplication.vue]: Corrected.

**Review by tmszi — APPROVED** [review]

**Pull request merged**